airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -73,191 +73,129 @@
-            connectionUpdaterInput.getConnectionId());
-      } else if (workflowState.isDeleted()) {
-        // Stop the runs
-        final ConnectionDeletionInput connectionDeletionInput = new ConnectionDeletionInput(connectionUpdaterInput.getConnectionId());
-        runMandatoryActivity((input) -> connectionDeletionActivity.deleteConnection(input), connectionDeletionInput);
-        return;
-      } else if (workflowState.isCancelled() || workflowState.isCancelledForReset()) {
-        runMandatoryActivity((input) -> jobCreationAndStatusUpdateActivity.jobCancelled(input),
-            new JobCancelledInput(
-                maybeJobId.get(),
-                maybeAttemptId.get(),
-                FailureHelper.failureSummaryForCancellation(maybeJobId.get(), maybeAttemptId.get(), failures, partialSuccess)));
-        resetNewConnectionInput(connectionUpdaterInput);
-      } else if (workflowState.isFailed()) {
-        reportFailure(connectionUpdaterInput);
-      } else {
-        // report success
-        reportSuccess(connectionUpdaterInput);
+      if (workflowState.isCancelled()) {
+        reportCancelledAndContinueWith(false, connectionUpdaterInput);
       }
-      continueAsNew(connectionUpdaterInput);
+
     } catch (final Exception e) {
       log.error("The connection update workflow has failed, will create a new attempt.", e);
-
-      reportFailure(connectionUpdaterInput);
-      continueAsNew(connectionUpdaterInput);
+      reportFailure(connectionUpdaterInput, null);
+      prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
     }
   }
 
-  private void reportSuccess(final ConnectionUpdaterInput connectionUpdaterInput) {
+  private void generateSyncWorkflowRunnable(ConnectionUpdaterInput connectionUpdaterInput) {
+    scopedForCancellationSyncWorkflow = Workflow.newCancellationScope(() -> {
+      connectionId = connectionUpdaterInput.getConnectionId();
+
+      if (connectionUpdaterInput.getWorkflowState() != null) {
+        workflowState = connectionUpdaterInput.getWorkflowState();
+      }
+
+      workflowState.setResetConnection(connectionUpdaterInput.isResetConnection());
+
+      Duration timeToWait = getTimeToWait(connectionUpdaterInput.getConnectionId());
+
+      Workflow.await(timeToWait,
+          () -> skipScheduling() || connectionUpdaterInput.isFromFailure());
+
+      val needToProcessSync = !workflowState.isUpdated() && !workflowState.isDeleted();
+
+      if (needToProcessSync) {
+        jobId = getOrCreateJobId(connectionUpdaterInput);
+
+        attemptId = createAttemptId(jobId);
+
+        final GeneratedJobInput jobInputs = getJobInput();
+
+        reportJobStarting();
+        StandardSyncOutput standardSyncOutput = null;
+        try {
+          standardSyncOutput = runChildWorkflow(jobInputs);
+
+          checkifFailedAndSetFailStatus(standardSyncOutput);
```

Comment:
```suggestion
          workflowState.setFailed(getFailStatus(standardSyncOutput));
```
Based on the names it's hard to tell why `checkifFailedAndSetFailStatus` vs `reportFailure` would be called. These state modifications within functions make it harder to tell what's happening at the higher level imo.

Reply:
Done